src/app/admin-panel/admin-panel/students/students.component.ts

```diff
@@ -1,15 +1,121 @@
 import { Component, OnInit } from '@angular/core';
+import { StudentsService } from '../../../services/admin-students.service';
+import { Student } from '../../../models/students.model';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import {Class} from '../../../models/classesForStudents';
 
 @Component({
   selector: 'app-students',
   templateUrl: './students.component.html',
-  styleUrls: ['./students.component.scss']
+  styleUrls: ['./students.component.scss'],
+  providers: [StudentsService]
 })
 export class StudentsComponent implements OnInit {
 
-  constructor() { }
+  classes: Class[];
+  students: Student[];
+  avatars: any[] = [];
+  newStudent: Student;
+  isNew: boolean;
+  cols: any[];
+  selectedClass: number = 0;
+  displayForm: boolean;
+  constructor(private service_: StudentsService, private formBuilder: FormBuilder) {
+    this.students = new Array<Student>();
+  }
+
+  addStudentForm: FormGroup;
 
   ngOnInit() {
+    this.service_.getClasses().subscribe(
+      data => {
+        this.classes = data as any;
+      });
+
+    this.loadStudents(1);
+
+    this.cols = [
+      { field: 'firstname', header: 'Ім\'я' },
+      { field: 'lastname', header: 'Прізвище' },
+      { field: 'patronymic', header: 'По-батькові' },
+      { field: 'classe', header: 'Клас' },
+      { field: 'dateOfBirth', header: 'Дата народження' },
+      { field: 'email', header: 'Email' },
+      { field: 'phone', header: 'Номер телефону' },
+      { field: 'login', header: 'Логін' },
+    ];
+
+    this.addStudentForm = this.formBuilder.group({
+      firstname: ['', Validators.required],
+      lastname: ['', Validators.required],
+      patronymic: ['', Validators.required],
+      classId: ['', Validators.required],
+      dateOfBirth: ['', Validators.required],
+      email: [''],
+      phone: [''],
+      login: ['', Validators.required],
+      avatar: [''],
+      selectClass: ['']
+    });
+
+    this.newStudent = new Student('', '', '', 0, '', '', '', '', 0);
```

Comment:
no need to pass default values into constructor

Reply:
Changed